app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java

```diff
@@ -87,4 +88,28 @@ public static void displayPopupWindow(View anchorView, Context context, View pop
         popup.showAsDropDown(anchorView);
     }
 
+    /**
+     * A snack bar which has an action button which on click dismisses the snackbar and invokes the listener passed
+     * @param view
+     * @param messageResourceId
+     * @param retryButtonResourceId
+     * @param onClickListener
+     */
+    public static void showDismissibleSnackBar(View view, int messageResourceId,
+        int retryButtonResourceId, View.OnClickListener onClickListener) {
```

Comment:
it would be better if we could keep the variable name generic. the button text could be anything

Reply:
Agreed, on it